[SDK-4819] Add log level descriptions for Android SDK (#7624)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/android/initial_sdk_setup/additional_customization_and_configuration.md

-keep class com.braze.** { *; }
```

## Logging

By default, the the Braze Android SDK log level is set to `INFO`. You can [suppress these logs](#suppressing-logs) or [set a different log level](#enabling-logs-enabling-logs), such as `VERBOSE`, `DEBUG`, or `WARN`.

### Enabling logs {#enabling-logs}

To help troubleshoot issues in your app, or ensure fast turnaround times with Braze Support, you'll want to enable verbose logs for the SDK. When you send verbose logs to Braze Support, ensure they begin as soon as you launch your application and end far after your issue occurs.

Keep in mind, verbose logs are only intended for your development environment, so you'll want to disable them before releasing your app.

{% alert important %}
Enable verbose logs before any other calls in `Application.onCreate()` to ensure your logs are as complete as possible.
{% endalert %}

{% tabs %}

> {% tab Braze SDK %}

Reviewer: ```suggestion
{% tab Application %}
```

> To enable logs in the Braze Android SDK, add the following to your `TODO` file:

Reviewer: ```suggestion
To enable logs in the Braze Android SDK, add the following to your applications `onCreate()` method _before_ any other methods.
```

> To enable logs in the Braze Android SDK, add the following to your `TODO` file:

Reviewer: @vanessaland , could you let me know where in the Braze SDK this should be set? Thanks!

{% tab braze.xml %}
To enable logs in the `braze.xml`, add the following to your file:

```xml
<integer name="com_braze_logger_initial_log_level">LOG_LEVEL_VALUE</integer>
```

Replace `LOG_LEVEL_VALUE` with the value of the log level you'd like to enable.

| Constant    | Value          | Description                                                               |
|-------------|----------------|---------------------------------------------------------------------------|
| `VERBOSE`   | 2              | Logs the most detailed messages for debugging and development.            |
| `DEBUG`     | 3              | Logs descriptive messages for debugging and development.                  |
| `INFO`      | 4              | Logs informational messages for general highlights.                       |
| `WARN`      | 5              | Logs warning messages for identifying potentially harmful situations.     |
| `ERROR`     | 6              | Logs error messages for indicating application failure or serious issues. |
| `ASSERT`    | 7              | Logs assertion messages when conditions are false during development.     |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

{% alert note %}

> If you set `LOG_LEVEL_VALUE` to a number `>=` these values, the logs will be forwarded to Android's default [`Log`](https://developer.android.com/reference/android/util/Log) method. If the number is `<` any of these values, the logs will be discarded.

Author: This message was intended to tell the user that their log levels are inclusive to the logs with values equal to or above the set log level. Every log has a value and the set log level is to set what severity of logs I want to see.

For example, if I set my application's log level to `WARN`/`5`, I will be able to see logs of status `5`, `6`, `7`, and any logs of `4`, `3`, `2` will be dropped.

As another example, if I set my application's log level to `VERBOSE`/`2` (the most detailed level), I will be able to see all the logs of all values, `2`, `3`, `4`, `5`, `6`, and `7`.

```suggestion
Logs at levels `>=` to your set `LOG_LEVEL_VALUE` will be forwarded to Android's default [`Log`](https://developer.android.com/reference/android/util/Log) method. Any logs `<` your set `LOG_LEVEL_VALUE` will be discarded.
```

Author: We might also want to refactor this note and put it outside of the tabbing, since it's relevant to either way setting the log level. Let me know what you think!

{% endalert %}

Your code should be similar to the following:

```xml
<integer name="com_braze_logger_initial_log_level">2</integer>
```
{% endtab %}
{% endtabs %}

### Verifying verbose logs

To verify that your logs are set to `VERBOSE`, check if `V/Braze` occurs somewhere in your logs. If it does, then verbose logs have been successfully enabled. For example:

```
2077-11-19 16:22:49.591 ? V/Braze v9.0.01 .bo.app.d3: Request started
```

### Suppressing logs

> The default log level for the Braze Android SDK is `INFO`. To suppress all logs for the Braze Android SDK, call `BrazeLogger.SUPPRESS` in your `TODO` file.

Reviewer: @vanessaland , could you let me know where in the Braze SDK this should be set? Thanks!

> The default log level for the Braze Android SDK is `INFO`. To suppress all logs for the Braze Android SDK, call `BrazeLogger.SUPPRESS` in your `TODO` file.

Reviewer: ```suggestion
The default log level for the Braze Android SDK is `INFO`. To suppress all logs for the Braze Android SDK, call `BrazeLogger.SUPPRESS` in your application's `onCreate()` method _before_ any other methods.
```

{% tab braze.xml %}
To enable logs in the `braze.xml`, add the following to your file:

```xml
<integer name="com_braze_logger_initial_log_level">LOG_LEVEL_VALUE</integer>
```

Replace `LOG_LEVEL_VALUE` with the value of the log level you'd like to enable.

| Constant    | Value          | Description                                                               |
|-------------|----------------|---------------------------------------------------------------------------|
| `VERBOSE`   | 2              | Logs the most detailed messages for debugging and development.            |
| `DEBUG`     | 3              | Logs descriptive messages for debugging and development.                  |
| `INFO`      | 4              | Logs informational messages for general highlights.                       |
| `WARN`      | 5              | Logs warning messages for identifying potentially harmful situations.     |
| `ERROR`     | 6              | Logs error messages for indicating application failure or serious issues. |
| `ASSERT`    | 7              | Logs assertion messages when conditions are false during development.     |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

{% alert note %}

> If you set `LOG_LEVEL_VALUE` to a number `>=` these values, the logs will be forwarded to Android's default [`Log`](https://developer.android.com/reference/android/util/Log) method. If the number is `<` any of these values, the logs will be discarded.

Reviewer: ```suggestion
Any logs at a level `>=` to your set `LOG_LEVEL_VALUE` will be forwarded to Android's default [`Log`](https://developer.android.com/reference/android/util/Log) method. Any logs `<` your set `LOG_LEVEL_VALUE` will be discarded.
<br><br>
For example, if `LOG_LEVEL_VALUE` is set to `2`,  logs at levels `2`, `3`, `4`, `5`, `6`, and `7` will be forwarded to the 'Log' method.
```
